### PR TITLE
Add keymap and settings for dactyl_manuform 6x6_5 thumb

### PR DESCRIPTION
**Pull request by DuMaM**
## Summary
Dactyl manuform 6x6 with 5 key thumb cluster and stm32 support.

## Description
I was working for some time on my own dactyl manuform 6x6 with 5 key thumb cluster.
I was quite painful to maintain it and apply all needed changes.
1. I'm using black pill for my builds
2. Personalny I don't use standard layout to used in 6x6 manuform
I also want to use a qmk configurator in feature for my personal usage.

That's why I added support for blackpill for dactyl manuform 6x6, I also do smal refactor and extracted a promicro settings to separate folder (not breaking change). I also converted default layout to default 5 thumb based, and added my personal one.
BTW. This solution can be used to clean up 5x6_5, 5x6_2_5 and 5x6 :wink:.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR
None

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/handwired/dactyl_manuform/6x6/blackpill_f411/rules.mk [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+BLUETOOTH_ENABLE = no       # Enable Bluetooth
+AUDIO_ENABLE = no           # Audio output
+
+
+# Build Options
+#   change yes to no to disable
+#
+KEYBOARD_SHARED_EP = yes
```
Comment (drashna) [inline keyboards/handwired/dactyl_manuform/6x6/blackpill_f411/rules.mk:28]:
```suggestion
```

File: keyboards/handwired/dactyl_manuform/6x6/config.h [hunk 2]
```diff
@@ -20,21 +20,14 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
 #include "config_common.h"
 
-#define PRODUCT_ID 0x3636
-#define DEVICE_VER 0x0001
-#define PRODUCT    Dactyl-Manuform (6x6)
+#define PRODUCT_ID  0x3636
+#define DEVICE_VER  0x0001
+#define PRODUCT     "Dactyl-Manuform (6x6)"
+#define DESCRIPTION "SplitKeyboard for messed up bones"
```
Comment (drashna) [inline keyboards/handwired/dactyl_manuform/6x6/config.h:26]:
```suggestion
#define PRODUCT     Dactyl-Manuform (6x6)
```

File: keyboards/handwired/dactyl_manuform/6x6/keymaps/default_5_thumb/keymap.c [hunk 3]
```diff
@@ -0,0 +1,47 @@
+// #include QMK_KEYBOARD_H
+#include QMK_KEYBOARD_H
+
+#define _QWERTY 0
+#define _LOWER 1
+#define _RAISE 2
```
Comment (drashna) [inline keyboards/handwired/dactyl_manuform/6x6/keymaps/default_5_thumb/keymap.c:6]: Should use an enum here instead
```suggestion
enum custom_layers {
    _QWERTY,
    _LOWER,
    _RAISE,
};
```

File: keyboards/handwired/dactyl_manuform/6x6/keymaps/default_5_thumb/keymap.c [hunk 4]
```diff
@@ -0,0 +1,47 @@
+// #include QMK_KEYBOARD_H
```
Comment (drashna) [inline keyboards/handwired/dactyl_manuform/6x6/keymaps/default_5_thumb/keymap.c:1]: Could you add a GPL2+ compatible license header here? 

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

Or 

https://github.com/qmk/qmk_firmware/blob/6b299fab6e6a3143c061b9ea005253361fcd524b/data/templates/base/keyboard.c#L1-L15

File: keyboards/handwired/dactyl_manuform/6x6/keymaps/dumam/keymap.c [hunk 5]
```diff
@@ -0,0 +1,56 @@
+// #include QMK_KEYBOARD_H
```
Comment (drashna) [inline keyboards/handwired/dactyl_manuform/6x6/keymaps/dumam/keymap.c:1]: Could you add a GPL2+ compatible license header here? 

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

Or 

https://github.com/qmk/qmk_firmware/blob/6b299fab6e6a3143c061b9ea005253361fcd524b/data/templates/base/keyboard.c#L1-L15

**Review by drashna — CHANGES_REQUESTED** [review]

**Comment by DuMaM** [comment]
@drashna Thanks for review. I applied your changes and also added GPL license header to default.c file.

File: keyboards/handwired/dactyl_manuform/6x6/blackpill_f411/rules.mk (hunk 1 above)
Comment (DuMaM) [reply keyboards/handwired/dactyl_manuform/6x6/blackpill_f411/rules.mk]: Man, this fixed my problems with keyboard detection after walking up. Big thanks :star_struck:

File: keyboards/handwired/dactyl_manuform/6x6/config.h (hunk 2 above)
Comment (DuMaM) [reply keyboards/handwired/dactyl_manuform/6x6/config.h]: done

File: keyboards/handwired/dactyl_manuform/6x6/keymaps/default_5_thumb/keymap.c (hunk 3 above)
Comment (DuMaM) [reply keyboards/handwired/dactyl_manuform/6x6/keymaps/default_5_thumb/keymap.c]: done

File: keyboards/handwired/dactyl_manuform/6x6/keymaps/default_5_thumb/keymap.c (hunk 4 above)
Comment (DuMaM) [reply keyboards/handwired/dactyl_manuform/6x6/keymaps/default_5_thumb/keymap.c]: done

File: keyboards/handwired/dactyl_manuform/6x6/keymaps/dumam/keymap.c (hunk 5 above)
Comment (DuMaM) [reply keyboards/handwired/dactyl_manuform/6x6/keymaps/dumam/keymap.c]: done

File: keyboards/handwired/dactyl_manuform/6x6/blackpill_f411/rules.mk (hunk 1 above)
Comment (DuMaM) [reply keyboards/handwired/dactyl_manuform/6x6/blackpill_f411/rules.mk]: done

File: keyboards/handwired/dactyl_manuform/6x6/blackpill_f411/rules.mk [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = yes        # Console for debug
+COMMAND_ENABLE = yes        # Commands for debug and configuration
+NKRO_ENABLE = no            # USB Nkey Rollover
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+BLUETOOTH_ENABLE = no       # Enable Bluetooth
```
Comment (drashna) [inline keyboards/handwired/dactyl_manuform/6x6/blackpill_f411/rules.mk:24]:
```suggestion
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/handwired/dactyl_manuform/6x6/blackpill_f411/rules.mk [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+
+# Build Options
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
+MOUSEKEY_ENABLE = yes       # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = yes        # Console for debug
+COMMAND_ENABLE = yes        # Commands for debug and configuration
+NKRO_ENABLE = no            # USB Nkey Rollover
```
Comment (noroadsleft) [inline keyboards/handwired/dactyl_manuform/6x6/blackpill_f411/rules.mk:18]:
```suggestion
NKRO_ENABLE = no            # Enable N-Key Rollover
```

File: keyboards/handwired/dactyl_manuform/6x6/promicro/readme.md [hunk 8]
```diff
@@ -0,0 +1,3 @@
+# Pro Micro onekey
```
Comment (noroadsleft) [inline keyboards/handwired/dactyl_manuform/6x6/promicro/readme.md:1]: This file should be updated for Pro Micro builds, or deleted. (I'd prefer the former.)

**Review by noroadsleft — CHANGES_REQUESTED** [review]

File: keyboards/handwired/dactyl_manuform/6x6/promicro/readme.md (hunk 8 above)
Comment (DuMaM) [reply keyboards/handwired/dactyl_manuform/6x6/promicro/readme.md]: done

**Review by drashna — APPROVED** [review]

**Review by noroadsleft — APPROVED** [review]

**Comment by noroadsleft** [comment]
Thanks!

**Review by noroadsleft — COMMENTED** [review]
edit: posted to wrong PR, ignore

**Pull request merged**